flashinfer/utils.py

```diff
@@ -1052,27 +1073,44 @@ def wrapper(*args, **kwargs):
                     # Assume all tensors are on the same device/capability
                     major, minor = get_compute_capability(tensor_arg.device)
                     capability = major * 10 + minor
-
                 if not has_backend_choices() and common_check is None:
                     raise ValueError(
                         f"Invalid @backend_requirement decorator usage: no backend choices and no common_check for {func.__name__}"
                     )
 
                 if has_backend_choices():
-                    if not is_backend_supported(backend, capability):
-                        extra = f" with capability {capability}" if capability else ""
-                        raise BackendSupportedError(
-                            f"{func.__name__} does not support backend '{backend}'{extra}"
-                        )
+                    if backend == "auto":
+                        if not suitable_auto_backends(**kwargs_with_defaults):
+                            raise BackendSupportedError(
+                                f"No suitable auto backends found for {func.__name__}"
+                            )
+                    else:
+                        if not is_backend_supported(backend, capability):
+                            extra = (
+                                f" with capability {capability}" if capability else ""
+                            )
+                            raise BackendSupportedError(
+                                f"{func.__name__} does not support backend '{backend}'{extra}"
+                            )
+                        if not _is_problem_size_supported(**kwargs_with_defaults):
+                            raise ValueError(
+                                f"Problem size is not supported for {func.__name__}"
+                            )
                 else:
```

Comment:
Maybe add a comment here "If the function doesnt have backends (i.e., there is only 1, implicit backend), run the following checks.